Care-CureHospital/WebAppPatient/Validation/MedicalRecordValidation.cs

```diff
@@ -0,0 +1,253 @@
+﻿using Backend;
+using Backend.Model.AllActors;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using WebAppPatient.Dto;
+
+namespace WebAppPatient.Validation
+{
+    public class MedicalRecordValidation
+    {
+        private Regex usernameRegex = new Regex(@"^[A-Za-zŠšĐđŽžČčĆć][A-Za-z0-9ŠšĐđŽžČčĆć]*$");
+        private Regex lettersOnlyRegex = new Regex(@"^[A-ZŠĐŽČĆ][A-Za-zŠšĐđŽžČčĆć ]*$");
+        private Regex jmbgRegex = new Regex(@"^[0-9]{13}$");
+        private Regex personalCardRegex = new Regex(@"^[0-9]{9}$");
+        private Regex healthInsuranceCardRegex = new Regex(@"^[0-9]{11}$");
+        private Regex contactNumberRegex = new Regex(@"^[+]*[(]{0,1}[0-9]{1,4}[)]{0,1}[-\s\./0-9]*$");
+        private Regex eMailRegex = new Regex(@"^[\w-\.]+@([\w-]+\.)+[\w-]{2,4}$");
+
+        public MedicalRecordValidation() { }
+
+        public bool ValidateMedicalRecord(MedicalRecordDto medicalRecordDto)
+        {
+            if (!ValidateUsername(medicalRecordDto.Patient.Username) || !ValidatePassword(medicalRecordDto.Patient.Password, medicalRecordDto.ConfirmedPassword)
+                || !ValidateLettersOnly(medicalRecordDto) || !ValidateJmbg(medicalRecordDto.Patient.Jmbg) || !ValidateIdentityCard(medicalRecordDto.Patient.IdentityCard)
+                || !ValidateHealthInsuranceCard(medicalRecordDto.Patient.HealthInsuranceCard) || !ValidateDateOfBirth(medicalRecordDto.Patient.DateOfBirth)
+                || !ValidateContactNumber(medicalRecordDto.Patient.ContactNumber) || !ValidateEMail(medicalRecordDto.Patient.EMail) || !EmptyAddress(medicalRecordDto))
+            {
+                return false;
+            }
+            else
+            {
+                return true;
+            }
+        }
+
+        private bool ValidateUsername(string username)
+        {
+            try
+            {
+                Match match = usernameRegex.Match(username);
+
+                if (string.IsNullOrEmpty(username))
+                {
+                    return false;
```

Comment:
You can extract regex string validation from every of this validation methods as separate method. Something like this:

private bool validateString(string input, Regex reg){
         try
            {
                Match match = reg.Match(input);
                return (!string.IsNullOrEmpty(input) && match.Success)
            }
            catch
            {
                return false;
            }
    }